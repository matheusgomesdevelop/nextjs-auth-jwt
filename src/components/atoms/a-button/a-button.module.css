.a_button {
    width: 100%;
    height: 44px;
    padding: 10px 18px;
    background: var(--gradient-primary-60050090-deg);
    box-shadow: var(--gradient-primary-60050090-deg-box-shadow);
    border: 1px solid var(--gradient-primary-60050090-deg);
    border-radius: 8px;
    font-size: var(--fs-text-md);
    font-weight: var(--fw-semibold);
    color: var(--white);
    display: flex;
    justify-content: center;
}