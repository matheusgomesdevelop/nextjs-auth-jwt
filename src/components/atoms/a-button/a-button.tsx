import { Button, ButtonProps } from "primereact/button";

import styles from "./a-button.module.css";

export const A_Button = ({ children, ...props }: ButtonProps) => {
  return (
    <Button
      data-testid="a-button"
      {...props}
      pt={{
        root: {
          className: styles.a_button,
        },
      }}
    >
      {children}
    </Button>
  );
};
