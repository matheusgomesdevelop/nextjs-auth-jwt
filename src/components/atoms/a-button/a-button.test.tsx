import { render, screen } from "@testing-library/react";

// Atoms
import { AButton } from "./a-button";

describe("Deve renderizar o a-button corretamente", () => {
  it("Deve renderizar o botÃ£o com o texto", () => {
    // Arrange
    const text = "a-button";
    render(<AButton>{text}</AButton>);

    // Act
    const get_button = screen.getByTestId("a-button");

    // Assert
    expect(get_button).toBeInTheDocument();
    expect(get_button.textContent).not.toHaveLength(0);
  });
});
