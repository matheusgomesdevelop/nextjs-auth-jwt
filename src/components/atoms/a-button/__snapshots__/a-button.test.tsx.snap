// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o a-button corretamente Deve preservar a estrutura visual do bot√£o 1`] = `
<button
  className="p-button p-component a_button"
  data-pc-name="button"
  data-pc-section="root"
  data-testid="a-button"
  disabled={false}
>
  a-button
</button>
`;
